### ISSUE TITLE
Compile fails for patches saved with hyphen in the filename

### ISSUE THREAD

**Issue by ndonald2** [plugdata-team/plugdata#427]
## System Info
**OS**: macOS Catalina (10.15.7)
**plugdata version**: v0.6.4

Should be a standard installation in every way I can think of

![Screen Shot 2023-01-24 at 12 55 12](https://user-images.githubusercontent.com/1433497/214395228-252f7c38-4b45-4137-b022-a2d7394c903f.png)

## Issue Description

**Expected Behavior**
Compiling a patch saved to a file that contains a hyphen in the name should succeed

**Actual Behavior**
Compiling a patch saved to a file that contains a hyphen in the name fails with missing build rule issues

**Repro Steps**

1. Create any compilable patch
2. Save it to a file named "my-patch.pd"
3. Go to Settings -> Compile...
4. Select the Daisy tab
5. Select any board and export type "Binary"
6. Click Export and choose a location/filename for the export
7. Click Save

At this point the logs will show an error saying there is no rule to build the object file `HeavyDaisy_my-patch.o`. I would expect this to work given hyphens in filenames are a somewhat common convention and perfectly safe in all major desktop filesystems. Or if there is some sort of unfixable limitation around this in the toolchain, a better error/warning would be preferable.

Example log output:

```
--> Generating C

=== Signal Order ===
__phasor_k~f {frequency:220.0 phase:0.0}
__var_k~f {k:0.5 step:0 reverse:False}
__sub~f
__abs~f
__var_k~f {k:0.25 step:0 reverse:False}
__sub~f
__var_k~f {k:6.283185307179586 step:0 reverse:False}
__mul~f
__mul~f
__mul~f
__mul~f
__var_k~f {k:0.007833333333333 step:0 reverse:False}
__var_k~f {k:-0.166666666666667 step:0 reverse:False}
__fma~f
__fma~f
__add~f
__add~f
AVX: 58.0 cycles / 7.25 cycles per frame
     512 frames @ 1000MHz >= 3.71us
SSE: 55.0 cycles / 13.75 cycles per frame
     512 frames @ 1000MHz >= 7.04us
CPU% #Objs Object Type      Performance
==== ===== ================ ===========
22%  1     __phasor_k~f     Counter({'avx': 13, 'sse': 10})
 4%  5     __var_k~f        Counter({'avx': 2.5, 'sse': 2.5})
10%  2     __sub~f          Counter({'avx': 6, 'sse': 6})
 0%  1     __abs~f          Counter({'avx': 0.5, 'sse': 0.5})
34%  4     __mul~f          Counter({'avx': 20, 'sse': 20})
17%  2     __fma~f          Counter({'avx': 10, 'sse': 10})
10%  2     __add~f          Counter({'avx': 6, 'sse': 6})
--> Generating Daisy module
Total compile time: 102.91ms
Compiling.../Users/kipschwinger/Library/plugdata/Toolchain/bin/arm-none-eabi-gcc -c -mcpu=cortex-m7 -mthumb -mfpu=fpv5-d16 -mfloat-abi=hard -DUSE_HAL_DRIVER -DSTM32H750xx -DHSE_VALUE=16000000  -DCORE_CM7 -DSTM32H750IB -DARM_MATH_CM7 -DUSE_FULL_LL_DRIVER -include stm32h7xx.h -I../../libDaisy -I../../libDaisy/src/ -I../../libDaisy/src/sys -I../../libDaisy/src/usbd -I../../libDaisy/src/usbh -I../../libDaisy/Drivers/CMSIS/Include/ -I../../libDaisy/Drivers/CMSIS/DSP/Include -I../../libDaisy/Drivers/CMSIS/Device/ST/STM32H7xx/Include -I../../libDaisy/Drivers/STM32H7xx_HAL_Driver/Inc/ -I../../libDaisy/Middlewares/ST/STM32_USB_Device_Library/Core/Inc -I../../libDaisy/Middlewares/ST/STM32_USB_Host_Library/Core/Inc -I../../libDaisy/Middlewares/ST/STM32_USB_Host_Library/Class/MSC/Inc -I../../libDaisy/core/ -I../../libDaisy/Middlewares/Third_Party/FatFs/src -O2 -Wall -Wno-missing-attributes -fasm -fdata-sections -ffunction-sections -Wno-stringop-overflow -g -ggdb -MMD -MP -MF"build/startup_stm32h750xx.d" -std=gnu11 -Wa,-a,-ad,-alms=build/startup_stm32h750xx.lst ../../libDaisy/core/startup_stm32h750xx.c -o build/startup_stm32h750xx.o
make: *** No rule to make target 'build/HeavyDaisy_my-patch.o', needed by 'build/Heavy_my-patch.elf'.  Stop.
make: *** Waiting for unfinished jobs....

```

**Comment by timothyschoen** [comment]
Interesting bug, no idea what's causing this, but a simple solution could be to test for "-" in the patch name, and rename that? I wonder if this bug also happens when you set a custom project name. I'll test this out soon!

Thanks for reporting!

**Comment by dromer** [comment]
It seems that `make` isn't too happy with such files. I think the easiest solution is to make sure that the name supplied to the `-n` argument of hvcc has all dashes replaced with underscores.

Maybe just a `std::replace(name.begin(), name.end(), '-', '_');` will do it?

**Comment by dromer** [comment]
Ok that is `name = name.replaceCharacter('-', '_');`

I will prepare a fix for this.

**Comment by timothyschoen** [comment]
Should be fixed by e479356
